File: Selenium_Automation_Tests/pages/dashboard_page.py
"""
Dashboard Page Object
"""

from selenium.webdriver.common.by import By
from pages.base_page import BasePage
from utils.config import Config


class DashboardPage(BasePage):
    """Dashboard page object model"""
    
    # Locators
    DASHBOARD_HEADER = (By.XPATH, "//h1[contains(text(), 'Dashboard') or contains(text(), 'Welcome')]")
    USER_MENU = (By.CSS_SELECTOR, ".user-menu, [data-testid='user-menu']")
    LOGOUT_BUTTON = (By.XPATH, "//button[contains(text(), 'Logout') or contains(text(), 'Sign Out')]")
    EVENTS_LINK = (By.LINK_TEXT, "Events")
    BOOKINGS_LINK = (By.LINK_TEXT, "My Bookings")
    
    def __init__(self, driver):
        super().__init__(driver)
        self.url = Config.get_url('/dashboard')
    
    def is_on_dashboard_page(self):
        """Check if on dashboard page"""
        return self.is_element_visible(self.DASHBOARD_HEADER, timeout=10) or "dashboard" in self.get_current_url().lower()
    
    def logout(self):
        """Logout user"""
        self.click(self.USER_MENU)
        self.click(self.LOGOUT_BUTTON)
    
    def navigate_to_events(self):
        """Navigate to events page"""
        self.click(self.EVENTS_LINK)
    
    def navigate_to_bookings(self):
        """Navigate to bookings page"""
        self.click(self.BOOKINGS_LINK)

File: Java_Selenium_Automation/src/test/java/com/aiu/pages/DashboardPage.java
package com.aiu.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;

/**
 * DashboardPage - Page Object for Dashboard page
 */
public class DashboardPage extends BasePage {

    // Locators
    @FindBy(xpath = "//h1[contains(text(), 'Dashboard') or contains(text(), 'Welcome')]")
    private WebElement dashboardHeading;

    @FindBy(xpath = "//nav//a[contains(text(), 'Events')]")
    private WebElement eventsNavLink;

    @FindBy(xpath = "//nav//a[contains(text(), 'My Bookings')]")
    private WebElement bookingsNavLink;

    @FindBy(xpath = "//nav//a[contains(text(), 'Profile')]")
    private WebElement profileNavLink;

    @FindBy(xpath = "//button[contains(text(), 'Logout')] | //a[contains(text(), 'Logout')]")
    private WebElement logoutButton;

    @FindBy(xpath = "//div[contains(@class, 'notification')]")
    private WebElement notificationBadge;

    @FindBy(xpath = "//div[@class='stats' or contains(@class, 'dashboard-stats')]")
    private WebElement statsSection;

    public DashboardPage(WebDriver driver) {
        super(driver);
    }

    /**
     * Check if on dashboard page
     */
    public boolean isOnDashboardPage() {
        return getCurrentUrl().contains("/dashboard") || isDisplayed(dashboardHeading);
    }

    /**
     * Get dashboard heading text
     */
    public String getDashboardHeading() {
        return getText(dashboardHeading);
    }

    /**
     * Navigate to Events page
     */
    public void navigateToEvents() {
        click(eventsNavLink);
        logger.info("Navigated to Events page");
    }

    /**
     * Navigate to My Bookings page
     */
    public void navigateToBookings() {
        click(bookingsNavLink);
        logger.info("Navigated to My Bookings page");
    }

    /**
     * Navigate to Profile page
     */
    public void navigateToProfile() {
        click(profileNavLink);
        logger.info("Navigated to Profile page");
    }

    /**
     * Perform logout
     */
    public void logout() {
        click(logoutButton);
        waitForPageLoad();
        logger.info("Logged out successfully");
    }

    /**
     * Check if notification badge is displayed
     */
    public boolean isNotificationDisplayed() {
        return isDisplayed(notificationBadge);
    }

    /**
     * Check if stats section is displayed
     */
    public boolean isStatsDisplayed() {
        return isDisplayed(statsSection);
    }
}
